hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneDelete.java

```diff
@@ -0,0 +1,206 @@
+            "-skipTrash: option bypasses trash, if enabled, and immediately " +
+            "deletes <src>.\n" +
+            "-safely: option requires safety confirmation, if enabled, " +
+            "requires confirmation before deleting large directory with more " +
+            "than <hadoop.shell.delete.limit.num.files> files. Delay is " +
+            "expected when walking over large directory recursively to count " +
+            "the number of files to be deleted before the confirmation.\n";
+
+    private boolean skipTrash = false;
+    private boolean deleteDirs = false;
+    private boolean ignoreFNF = false;
+    private boolean safeDelete = false;
+    private boolean trailing = false;
+
+    @Override
+    protected void processOptions(LinkedList<String> args) throws IOException {
+      CommandFormat cf = new CommandFormat(
+          1, Integer.MAX_VALUE, "f", "r", "R", "skipTrash", "safely");
+      cf.parse(args);
+      ignoreFNF = cf.getOpt("f");
+      deleteDirs = cf.getOpt("r") || cf.getOpt("R");
+      skipTrash = cf.getOpt("skipTrash");
+      safeDelete = cf.getOpt("safely");
+    }
+
+    @Override
+    protected List<PathData> expandArgument(String arg) throws IOException {
+      try {
+        // handle trailing slash for symlinks
+        if (arg.endsWith(OZONE_URI_DELIMITER)) {
+          trailing = true;
+        }
+        return super.expandArgument(arg);
+      } catch (PathNotFoundException e) {
+        if (!ignoreFNF) {
+          throw e;
+        }
+        // prevent -f on a non-existent glob from failing
+        return new LinkedList<PathData>();
+      }
+    }
+
+    @Override
+    protected void processNonexistentPath(PathData item) throws IOException {
+      if (!ignoreFNF) {
+        super.processNonexistentPath(item);
+      }
+    }
+
+    @Override
+    protected void processPath(PathData item) throws IOException {
+      boolean isSymlink = false;
+      if (item.fs.supportsSymlinks()) {
+        isSymlink = item.fs.getLinkTarget(item.path) != item.path;
+      }
+      // support posix rm of symlink in addition to rm -r for directories
+      if (item.stat.isDirectory() && !deleteDirs && !isSymlink) {
+        throw new PathIsDirectoryException(item.toString());
+      }
+      // support posix symlink delete with trailing slash 'rm <path symlink>/'
+      // deletes contents of symlink bucket and retains symlink
+      Path path = item.path;
+      if (isSymlink && trailing) {
+        path = new Path(URI.create(path.toString() + OZONE_URI_DELIMITER));
+      }
+      // TODO: if the user wants the trash to be used but there is any
+      // problem (ie. creating the trash dir, moving the item to be deleted,
+      // etc), then the path will just be deleted because moveToTrash returns
+      // false and it falls thru to fs.delete.  this doesn't seem right
+      if (moveToTrash(item) || !canBeSafelyDeleted(item)) {
+        return;
+      }
```

Comment:
I'm trying to understand this TODO here.

I think in this case (where `skipTrash == false` but moving to trash has issues) we should just return error to the user. So that they either need to deal with that trash issue first (maybe some ACL issue), or set `-skipTrash`.

I don't think falling back to to `skipTrash` would be the intended user behavior, cause they could lose keys when they think those would be moved to the trash.

Also it looks like `moveToTrash()` is already throwing exception appropriately? Do we need extra handling here? If not we can just fix the comment.

Reply:
Looked into the moveToTrash() comment.

> just return error to the user. So that they either need to deal with that trash issue first (maybe some ACL issue), or set -skipTrash.

I suspect, like you that errors from creating the trash directory or renaming to trash will be handled properly with the exception thrown by trash - similar to 
```
bash-4.2$ ozone fs -rm ofs://om/tmp/tmpfile
rm: Failed to move to trash: ofs://om/tmp/tmpfile: User testuser2/scm@EXAMPLE.COM 
doesn't have DELETE permission to access key Volume:tmp Bucket:tmp Key:tmpfile. 
Consider using -skipTrash option
```

Here errors occurring from trash unable to create trash dir or renaming file are handled by exception handling and the file is not outright deleted by `fs.delete`.  File remains intact leaving the user to resolve the issue through permission settings or `-skipTrash`.

I'm looking to update the comment to reflect this.